components/Hyperlinks.js

```diff
@@ -1,7 +1,69 @@
 import gql from 'graphql-tag';
-import CircularProgress from '@material-ui/core/CircularProgress';
+import { Box, SvgIcon, CircularProgress } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
 import { useQuery } from '@apollo/react-hooks';
 
+const useStyles = makeStyles(theme => ({
+  root: {
+    padding: '12px 16px',
+    margin: '0 8px 8px 0',
+    background: theme.palette.secondary[50],
+    borderRadius: 8,
+    '& h1': {
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+      whiteSpace: 'nowrap',
+      margin: 0,
+      marginBottom: 10,
+      maxWidth: 'inherit',
+    },
+  },
+  url: {
+    color: theme.palette.secondary[300],
+    display: 'flex',
+    alignItems: 'center',
+    marginTop: 16,
+    '& a': {
+      color: theme.palette.secondary[300],
+      display: 'block',
+      whiteSpace: 'nowrap',
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+      textDecoration: 'none',
+      marginLeft: 8,
+    },
+  },
+  preview: {
+    display: 'flex',
+    borderLeft: `4px solid ${theme.palette.secondary[200]}`,
+    paddingLeft: 10,
+    paddingRight: 15,
+    minHeight: 40,
+    justifyContent: 'space-between',
+    '& .summary': {
+      color: theme.palette.secondary[500],
+      maxHeight: 40,
+      overflow: 'hidden',
+      margin: 0,
+      display: 'flex',
```

Comment:
This `flex` is not needed.
Instead, I think we can add `-webkit-line-clamp` (and other properties to make it work) https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp so that it displays `...` when text is truncated.